search/includes/classes/class-search.php

```diff
@@ -980,12 +980,8 @@ public function is_url_query_cacheable( string $url, $args ): bool {
 			return $is_cacheable;
 		}
 
-		foreach ( [ '_search', '_mget', '_doc' ] as $needle ) {
-			if ( wp_in( $needle, $url ) ) {
-				$is_cacheable = true;
-				break;
-			}
-		}
+		$is_cacheable = (bool) preg_match( '#/_(search|mget|doc)#', $url );
```

Comment:
While this is a cleaner solution, regex typically is slower. 

Using `hrtime()` for benchmarking:
With preg_match - 40167
Using old solution - 7417

Do you think the above is negligible enough to not affect the performance since it runs on every intercepted request?

Reply:
@rebeccahum  That's a good point, I obviously didn't measure. over how many iterations did you do the benchmarking?

Reply:
@rinatkhaziev Just 5-6 times and those were the averages.

Reply:
@rebeccahum https://replit.com/@rinatkhaziev/SpecializedAnguishedBruteforceprogramming#main.php

The old one is indeed more performant but the fluctuation on small repeats (where $i < 10) is wildly different, I got anywhere from 2 to 10 times slower. However, on 100000 iterations the difference is like 50%-200%, given that this is in milliseconds land  I think it should be fine?

Reply:
Scratch that, silly me forgot to reset the timer, the worst I could get is this:

Over 1 million iterations, the first one is the old approach.

```
string(10) "Elapsed ms"
float(277.970179)

string(10) "Elapsed ms"
float(363.893794)
```

So I think it's definitely fine :)